Add Go workflows

biinari opened the pull request:
https://fatsoma.atlassian.net/browse/PLAT-1203

Add reusable GitHub workflow for go tests and syntax / security checks.

Sample usage (once tagged with `v1`):

```yml
name: Go

on: [push]

jobs:
  syntax:
    uses: Fatsoma/reusable-actions/.github/workflows/go-syntax.yml@v1

  test:
    uses: Fatsoma/reusable-actions/.github/workflows/go-test.yml@v1
    with:
      enable-translate: true

  security:
    uses: Fatsoma/reusable-actions/.github/workflows/go-security.yml@v1
```

biinari committed:
Add Go workflows

Use with:

```yml
name: Go

on: [push]

jobs:
  syntax:
    uses: Fatsoma/reusable-actions/.github/workflows/go-syntax.yml@v1

  test:
    uses: Fatsoma/reusable-actions/.github/workflows/go-test.yml@v1
    with:
      enable-translate: true

  security:
    uses: Fatsoma/reusable-actions/.github/workflows/go-security.yml@v1
```

ednp commented on .github/workflows/go-syntax.yml:
```diff
@@ -0,0 +1,26 @@
+      - name: Go Format
+        run: |
+          GOFMT_OUTPUT=$(gofmt -l $(find . -path '*/vendor/*' -prune -o -name '*.go' -print))
+          if [ -n "$FMT_OUTPUT" ] ; then
```
Does this want to be:
```suggestion
          if [ -n "$GOFMT_OUTPUT" ] ; then
```

biinari replied:
Good spot

biinari committed:
Fix GOFMT_OUTPUT check

Fix spelling of GOFMT_OUTPUT variable when checking if the output was non-empty

biinari committed:
Add sample usage for go workflow to readme

biinari committed:
go-test: Add option enable-redis

Add option to enable a redis server in go-test workflow

biinari committed:
go-test: Install translation dependencies only if enable-translate

biinari committed:
go-test: Set empty TZ

Ensure we are not in a local timezone when running tests

biinari committed:
Switch to joschi/go-junit-report

jstemmer repo seems to have stagnated and fails to install for go 1.25

biinari committed:
remove gocov, use go coverage tools
